### DISCUSSION TITLE
Rating Graph for more than 1 user

### DISCUSSION THREAD

**Discussion by hape42** [hape42/dailygammon#118] [Ideas]
As I learned, there are also users who play with multiple users in the app. But since the rating is only saved with date & value, this of course does not yield a meaningful evaluation for the case.

In order to save space and to have the data in the iCloud, I saved it in the KeyValueStore. However, this no longer works with multiple users.

So I will store this in CoreData / iCloud. 

iCloud will of course be optional.

If I already tackle the whole topic, I will also replace the already outdated graphics library with a modern library.

When I describe this just now, I realize that this will be a somewhat larger project. 

If anyone has ideas that fit into this project I look forward to your input. 

Translated with www.DeepL.com/Translator (free version)

**Comment by kagsteiner** [comment]
If we change the rating graph to support multiple users, we probably should also the login screen to support multiple users? Then this would mean we store user/password locally for each of them. Is this a security issue? But would be nice to just select one of the users you’ve entered, I guess.

Concerning the graph itself: maybe it would be nice to see the best rating ever marked in a special way. Many DG users seem to be proud of their best rating and add it to their bio, so this might be valuable info.

The graph somehow looks dated, but I have no clue why, and what a “modern looking” graph should look like. :-(

Some such graphs in other apps support tapping somewhere and then showing what the value at the respective date was. But honestly I don’t see why this should be useful.

**Reply by hape42** [reply]
> If we change the rating graph to support multiple users, we probably should also the login screen to support multiple users? Then this would mean we store user/password locally for each of them. Is this a security issue? But would be nice to just select one of the users you’ve entered, I guess.

I don't think it's a lot of work. 
I don't have any security concerns, since the passwords are only stored in the app on exactly this device.

**Reply by hape42** [reply]
> Concerning the graph itself: maybe it would be nice to see the best rating ever marked in a special way. Many DG users seem to be proud of their best rating and add it to their bio, so this might be valuable info.

This is not a big problem. I have the data. You could output the high score and the low score.

**Comment by hape42** [comment]
done
